src/plugins/workspace/public/components/workspace_form/use_workspace_form.ts

```diff
@@ -107,34 +123,19 @@ export const useWorkspaceForm = ({
         return;
       }
 
-      onSubmit?.(
-        {
-          name: currentFormData.name!,
-          description: currentFormData.description,
-          color: currentFormData.color || '#FFFFFF',
-          features: currentFormData.features,
-          permissionSettings: currentFormData.permissionSettings as WorkspacePermissionSetting[],
-          selectedDataSourceConnections: currentFormData.selectedDataSourceConnections,
-        },
-        true
-      );
+      const submitFormData = getSubmitFormData(currentFormData);
+      const result = await onSubmit?.(submitFormData);
+      if (result?.success) {
+        defaultValuesRef.current = submitFormData;
+        setIsEditing(false);
+      }
     },
     [onSubmit, permissionEnabled]
   );
 
   const handleSubmitPermissionSettings = async (settings: WorkspacePermissionSetting[]) => {
     const currentFormData = getFormDataRef.current();
-    const result = await onSubmit?.(
-      {
-        name: currentFormData.name!,
-        description: currentFormData.description,
-        color: currentFormData.color || '#FFFFFF',
-        features: currentFormData.features,
-        permissionSettings: settings,
-        selectedDataSourceConnections: currentFormData.selectedDataSourceConnections,
-      },
-      false
-    );
+    const result = await onSubmit?.(getSubmitFormData(currentFormData, settings));
```

Comment:
```suggestion
    const result = await onSubmit?.({ ...getSubmitFormData(currentFormData), permissionSettings: settings });
```
Should be good enough to use spread and override.

Reply:
Thanks for your suggestion. The code has been updated